Src/Support/Google.Apis.Auth.Tests/OAuth2/UserCredentialTests.cs

```diff
@@ -0,0 +1,118 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+using Google.Apis.Auth.OAuth2;
+using Google.Apis.Auth.OAuth2.Flows;
+using Google.Apis.Auth.OAuth2.Requests;
+using Google.Apis.Auth.OAuth2.Responses;
+using Google.Apis.Util;
+using Google.Apis.Util.Store;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Google.Apis.Auth.Tests.OAuth2
+{
+    public class UserCredentialTests
+    {
+        [Fact]
+        public async Task UniverseDomain_Default()
+        {
+            var credential = new UserCredential(
+                FakeAuthorizationCodeFlow.Default, "userId", token: null, quotaProjectId: "quotaProject") as IGoogleCredential;
+
+            Assert.Equal(GoogleAuthConsts.DefaultUniverseDomain, credential.GetUniverseDomain());
+            Assert.Equal(GoogleAuthConsts.DefaultUniverseDomain, await credential.GetUniverseDomainAsync(default));
+        }
+
+        [Theory]
+        [InlineData(null)]
+        [InlineData(GoogleAuthConsts.DefaultUniverseDomain)]
+        public async Task WithUniverseDomain_Default(string universeDomain)
+        {
+            var credential = new UserCredential(
+                FakeAuthorizationCodeFlow.Default, "userId", token: null, quotaProjectId: "quotaProject") as IGoogleCredential;
+
+            var newCredential = credential.WithUniverseDomain(universeDomain);
+
+            Assert.Same(credential, newCredential);
+
+            Assert.Equal(GoogleAuthConsts.DefaultUniverseDomain, newCredential.GetUniverseDomain());
+            Assert.Equal(GoogleAuthConsts.DefaultUniverseDomain, await newCredential.GetUniverseDomainAsync(default));
+        }
+
+        [Fact]
+        public void WithUniverseDomain_Custom_Fails()
+        {
+            var credential = new UserCredential(
+                FakeAuthorizationCodeFlow.Default, "userId", token: null, quotaProjectId: "quotaProject") as IGoogleCredential;
+
+            Assert.Throws<InvalidOperationException>(() => credential.WithUniverseDomain("universeDomain"));
+        }
+
+        private class FakeAuthorizationCodeFlow : IAuthorizationCodeFlow
+        {
+            public static readonly FakeAuthorizationCodeFlow Default = new FakeAuthorizationCodeFlow();
+
+            public IAccessMethod AccessMethod => throw new NotImplementedException();
+
+            public IClock Clock => SystemClock.Default;
+
+            public IDataStore DataStore => throw new NotImplementedException();
+
+            public AuthorizationCodeRequestUrl CreateAuthorizationCodeRequest(string redirectUri)
```

Comment:
Use expression-bodied members for all of these? (I think it's fine for that to mean long lines, probably not even with line-breaks between the methods, for just a block of "stuff we haven't implemented.)